API/Documentation.md

```diff
@@ -0,0 +1,42 @@
+An API without an accompanying documentation practically doesn't even exist. Furthermore, without good, thorough documentation, no API can be good either. As the developer behind an API, you naturally know what you've made and how to use it. However, your clients don't. Documentation exists to transfer that knowledge to clients.
+
+Documentation writing requires thinking about what the API client needs to set up their environment, what they're able to do with the API, what to send in requests and which response scenarios to expect. All of this naturally requires providing lots of information and structuring it in a comprehensible way.
+
+Writing isn't easy and developers tasked with writing documentation sometimes do a poor job. Poorly-written documentation necessitates further communication and explanations from the API developer — imagine then what would happen if you had to explain your API to a thousand confused clients. Therefore, to save time in the long run, you have to invest enough time to write proper documentation from the start.
+
+## Essentials
+
+The following checklist lists documentation essentials. This is not an exhaustive list — APIs always have specifics which cannot be included in such lists, though you should document them too. A good rule of thumb is, if someone would ask about it, it should be documented.
+
+- **Overview** - Placed at the top of the documentation, the overview should explain: what the API does, architecture (e.g. REST), format (e.g. JSON:API or GraphQL) and should provide general information (base URL, response types, etc.).
+- **Format requirements** - If you use JSON:API, GraphQL or some other format, they typically come with requirements, like setting `Content-Type`/`Accept` headers, which the clients must abide by.
+- **CORS** - If applicable, document your CORS policy (allowed origins, methods, exposed headers, credentials, etc.).
+- **Authentication** - Applicable if endpoints are authenticated. Explain the authentication method(s) used (API key, cookies, JWT, etc.) and how to obtain credentials.
+- **Authorization** - If the outcome of requests depends on the user's permissions, then that should also be documented. If authorization logic varies between endpoints, each endpoint should explain its logic.
+- **Errors** - All APIs are subject to errors. Errors are roadblocks, preventing clients from doing what they need to do. And when a roadblock comes with either no explanation or a vague one (`PC LOAD LETTER`), frustration ensues.<br />
+Document <u>all</u> common errors and supply descriptions. Descriptions should explain what happened and what the client can do to remove the error. Document even the 5xx messages, no matter how obvious they seem.
+- **Changelog** - Major changes to the API should be documented for historical reference. It's an optional step if the API will be used by a limited number of users to whom you can communicate changes directly, though it's useful to have even then.
```

Comment:
Do we have an example of this on a project?

Reply:
On truck... https://api-truck-staging.byinfinum.co/v4/docs/#changelog
Dont know if we can show that as well...

Reply:
On one project, yes